addons/account/models/account_move_line.py

```diff
@@ -3389,6 +3389,16 @@ def _prepare_edi_vals_to_export(self):
         }
         return res
 
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
+        # Add total amount_currency to the read_group when view is grouped by currency_id only
+        res = super().read_group(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
+        if 'currency_id' in groupby:
+            for group_line in res:
+                lines = self.search(group_line.get('__domain'))
+                group_line['amount_currency'] = sum(lines.mapped('amount_currency'))
```

Comment:
```suggestion
                res = super().read_group(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
        if 'currency_id' not in groupby and 'amount_currency' in fields:
            for group_line in res:
                group_line['amount_currency'] = 0
```

I actually have another idea (JUAL's idea actually), which is to do like you did originally, meaning removing the `aggregator=None` and "hide" the results of `amount_currency` if `currency_id` is not in the groupby. What do you think of this idea ? I find it much simpler, and much less scary performance wise. Another good point for this idea is that when JUAL added the aggregator=None, the spreadsheet team was upset because they apparently could not anymore use properly the amount_currency. With this way of doing, it would become possible again using for example a context key.

Note: I think you should also check if `amount_currency` is in the fields

Reply:
I think that sounds good, but maybe we need to discuss with the PO since it means that when grouping by fields other than currency, 0.0 will be displayed. 
![image](https://github.com/user-attachments/assets/9bf6c917-146c-4dd0-a578-1f32ed4b53ce)

Reply:
Right I didn't think about that, is there not a way to hide it instead of providing the value 0 ?